pre_processing/risk_mapper.py

```diff
@@ -0,0 +1,124 @@
+            .reset_index()
+            .rename(columns={'index': 'cat', 0: 'name'}))
+    idx = pd.MultiIndex.from_tuples(cats.name.apply(_get_intervals_from_name),
+                                    names=['gestation_time', 'birth_weight'])
+    cats = cats['cat']
+    cats.index = idx
+    return cats
+
+def _get_lbwsg_exposed_unexposed_cats():
+    """
+    Get the exposed and unxeposed categories for Low Birth Weight and Short Gestation.
+    Exposed categories are those with either low birth weight (<2500g) or short gestation (<37wk).
+    Unexposed categories are the rest.
+    
+    Note: It would be better to enable passing specific cutoff values for different use cases.
+    (E.g. maybe we're interested in extreme preterm births (<27wk) rather than just preterm (<37wk).)
+    """
+    LBW_CUTOFF = 2500 #grams - standard LBW cutoff
+#     SG_CUTOFF = 37 #weeks - standard SG cutoff
+    SG_CUTOFF = 0 # We don't care about short gestation right now - want LBW exposure only
+    
+    # Get pd.Series that maps interval pairs to categories, and call .reset_index()
+    # to get a dataframe with a column for each interval, as well as a 'cat' column.
+    lbwsg = gbd_mapping.risk_factors.low_birth_weight_and_short_gestation
+    cats = _get_lbwsg_categories_by_interval(lbwsg.categories.to_dict())
+    cats = cats.reset_index()
+    
+    # Get the minimum and supremum of the birth weights and gestational ages for each category.
+    # (Note that it's a supremum not a maximum since the intervals are right-open.)
+    cats['min_birth_weight'] = cats['birth_weight'].apply(lambda x: x.left)
+    cats['supremum_birth_weight'] = cats['birth_weight'].apply(lambda x: x.right)
+    cats['min_gestation_time'] = cats['gestation_time'].apply(lambda x: x.left)
+    cats['supremum_gestation_time'] = cats['gestation_time'].apply(lambda x: x.right)
+    
+    exposed_unexposed_cats = {}
+    
+    # Note that since the intervals are left-closed and right-open, e.g. [2000,2500) and [2500,3000),
+    # we need to use <= for the exposed categories to get all the values < CUTOFF.
+    exposed_unexposed_cats['exposed'] = cats.loc[
+        (cats['supremum_birth_weight'] <= LBW_CUTOFF) |
+        (cats['supremum_gestation_time'] <= SG_CUTOFF),
+        'cat']
+    
+    exposed_unexposed_cats['unexposed'] = cats.loc[
+        (cats['min_birth_weight'] >= LBW_CUTOFF) &
+        (cats['min_gestation_time'] >= SG_CUTOFF),
+        'cat']
+    
+    # A category is 'partially exposed' if one of its intervals spans a cutoff.
+    # There *shouldn't* be any of these for the standard cutoff values.
+    exposed_unexposed_cats['partially_exposed'] = cats.loc[
+        ((cats['supremum_birth_weight'] > LBW_CUTOFF) & (cats['min_birth_weight'] < LBW_CUTOFF)) |
+        ((cats['supremum_gestation_time'] > SG_CUTOFF) & (cats['min_gestation_time'] < SG_CUTOFF)),
+        'cat']
+    
+    return exposed_unexposed_cats
+
+_risk_functions = {
```

Comment:
It doesn't look like `_risk_functions` is used anywhere besides `get_exposed_unexposed_cats_for_risk`. Consider nesting it in the function.

Reply:
Cool, thanks for the tip. I think I made it a module level variable so that the dictionary wouldn't have to get initialized every time the function is called, but I'm sure the efficiency gain is negligible in this case...